health-services/household/src/main/java/org/egov/household/validators/household/HHouseholdTypeChangeValidator.java

```diff
@@ -0,0 +1,107 @@
+package org.egov.household.validators.household;
+
+import lombok.extern.slf4j.Slf4j;
+import org.egov.common.models.Error;
+import org.egov.common.models.household.Household;
+import org.egov.common.models.household.HouseholdBulkRequest;
+import org.egov.common.models.household.HouseholdSearch;
+import org.egov.common.validator.Validator;
+import org.egov.household.repository.HouseholdRepository;
+import org.egov.tracer.model.CustomException;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.egov.common.utils.CommonUtils.*;
+import static org.egov.common.utils.CommonUtils.notHavingErrors;
+import static org.egov.common.utils.ValidatorUtils.getErrorForNonExistentEntity;
+import static org.egov.household.Constants.GET_ID;
+
+@Slf4j
+@Component
+@Order(value = 3)
+public class HHouseholdTypeChangeValidator implements Validator<HouseholdBulkRequest, Household> {
+
+    private final HouseholdRepository householdRepository;
+
+    public HHouseholdTypeChangeValidator(HouseholdRepository householdRepository) {
+        this.householdRepository = householdRepository;
+    }
+
+    @Override
+    public Map<Household, List<Error>> validate(HouseholdBulkRequest request) {
+        // Map to hold household entities and their error details
+        Map<Household, List<Error>> errorDetailsMap = new HashMap<>();
+        // Get the list of household entities from the request
+        List<Household> entities = request.getHouseholds();
+        // Get the class of the household entity
+        Class<?> objClass = getObjClass(entities);
+        // Get the method for fetching the ID of the entity
+        Method idMethod = getMethod(GET_ID, objClass);
```

Comment:
Models have getid method directly now, no need for reflections utils

Reply:
done